### ISSUE TITLE
Concurrency problem with this registration method？

### ISSUE THREAD

**Issue by 139446702556** [alibaba/nacos#6640]
public void register(String namespace, String group, ExecutorService executor) {
		if (!resourcesManager.containsKey(namespace)) {
        	synchronized(this) {
                lockers.put(namespace, new Object());
            }
        }
        final Object monitor = lockers.get(namespace);
        synchronized (monitor) {
            Map<String, Set<ExecutorService>> map = resourcesManager.get(namespace);
            if (map == null) {
            	map = new HashMap<String, Set<ExecutorService>>(8);
            	map.put(group, new HashSet<ExecutorService>());
				map.get(group).add(executor);
				resourcesManager.put(namespace, map);
				return;
			}
            if (!map.containsKey(group)) {
                map.put(group, new HashSet<ExecutorService>());
            }
            map.get(group).add(executor);
        }
    }

With each namespace in the lockers creating a new object at a time, there may be concurrency issues？

**Comment by brotherlu-xcq** [comment]
Yes, this codes is not thread safe. The first synchronized code is already not thread safe. this class is often used by Class init, so now it has not concurrency issue. 
due to it  is often used in init Class, not the runtime. So I think we can just use the synchronized to method not the method body.
like 
```java
public synchronized void register(String namespace, String group, ExecutorService executor) {
  xxx
}
```

**Comment by KomachiSion** [comment]
Where this register used?

If not used by multiple thread will not problem.

**Comment by 139446702556** [comment]
Ok, thank you，I'm just wondering what the use of a lock is at this point, and if it doesn't solve the concurrency problem, is the lock meaningless？

**Comment by 139446702556** [comment]
> Yes, this codes is not thread safe. The first synchronized code is already not thread safe. this class is often used by Class init, so now it has not concurrency issue.
> due to it is often used in init Class, not the runtime. So I think we can just use the synchronized to method not the method body.
> like
> 
> ```java
> public synchronized void register(String namespace, String group, ExecutorService executor) {
>   xxx
> }
> ```

Ok, thank you，I'm just wondering what the use of a lock is at this point, and if it doesn't solve the concurrency problem, is the lock meaningless？

**Comment by brotherlu-xcq** [comment]
@KomachiSion I find this method is used in static field of class now , It will init with Class load. It has not  concurrency issue now. need we fix it?

**Comment by 139446702556** [comment]
@brotherlu-xcq   Please tell me, I think the container in the ThreadPoolManager class is only used for the preservation of the mapping between the thread pool and the group, and is not used for other purposes. What is the role of this class?

**Comment by KomachiSion** [comment]
If no multiple thread used, I think has no need to add `synchronized`.

I remember this method only used in startup and startup workflow is an single thread.
